libs/designsystem/header-actions/src/header-actions.component.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  ContentChildren,
+  ElementRef,
+  HostBinding,
+  Input,
+  Renderer2,
+  ViewChild,
+} from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { ButtonComponent } from '@kirbydesign/designsystem/button';
+import { DropdownComponent, DropdownModule } from '@kirbydesign/designsystem/dropdown';
+
+@Component({
+  selector: 'kirby-header-actions',
+  standalone: true,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  imports: [CommonModule, DropdownModule],
+  templateUrl: './header-actions.component.html',
+  styleUrls: ['./header-actions.component.scss'],
+})
+export class HeaderActionsComponent implements AfterViewInit {
+  /*
+   * When the consumer sets the visibleActions input, we run all our initialization logic.
+   * The guard for this.buttons exist because when the component is instantiated,
+   * the ContentChild query has not yet run, and all the below functions assume that
+   * the buttons array is defined. When undefined, we defer the initialization to ngAfterViewInit().
+   */
+  @Input() set visibleActions(value: number) {
+    this._visibleActions = value;
+
+    if (!this.buttons) return;
```

Comment:
Is `this.buttons` undefined or should we check for `this.buttons.length === 0`? 🤔

Reply:
It is undefined but good suggestion, will rewrite to `this.buttons?.length === 0`